src/client/auth/mod.rs

```diff
@@ -349,6 +407,42 @@ impl Credential {
     }
 }
 
+/// Contains the first client message sent as part of the authentication handshake.
+pub(crate) enum ClientFirst {
+    Scram(ScramVersion, scram::ClientFirst),
+    X509(Command),
```

Comment:
Since the X.509 handshake is fairly simple, we don't have much of a need for a structured type for this, so I just used Command.